### PR TITLE
DUOS-842 [risk=no] Timestamp and Partial DAR Code bugfixes

### PR DESCRIPTION
**Pull request by JVThomas**
Addresses [DUOS-842](https://broadinstitute.atlassian.net/browse/DUOS-842)

PR adds create date and sort date to document records on GET requests, PR also generates partialDarCode for new drafts (v2 resource file). Additions address the current issue on the UI where requested DARs do not have partialDarCodes assigned on generation and does not have create and sort dates added to document returns, resulting in blank fields and occasional component breakage due to missing attributes.

As mentioned on the ticket, the PR is a companion piece to DUOS-681 since that PR utilizes the new v2 endpoints (which is how the bug was initially discovered), so testing alongside that ticket is advised.

Branch is also based off of DUOS-832, which is another companion piece to DUOS-681

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've followed [the instructions](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [x] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [x] I've updated the [FISMA documentation](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing
- [x] I've updated Swagger to reflect any API changes

In all cases:

- [x] Get two thumbsworth of review and PO signoff if necessary
- [x] Verify all tests go green
- [x] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java [hunk 1]
```diff
@@ -245,6 +245,8 @@ private Document createDocumentFromDar(DataAccessRequest d) {
         Document document = Document.parse(gson.toJson(d.getData()));
         document.put(DarConstants.DATA_ACCESS_REQUEST_ID, d.getId());
         document.put(DarConstants.REFERENCE_ID, d.getReferenceId());
+        document.put(DarConstants.CREATE_DATE, d.getCreateDate());
+        document.put(DarConstants.SORT_DATE, d.getCreateDate());
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java]: This line should be: 
```suggestion
        document.put(DarConstants.SORT_DATE, d.getSortDate());
```

Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java]: I think we also need to make a couple similar updates to `public DataAccessRequest insertDraftDataAccessRequest(User user, DataAccessRequest dar)`. Can you add these lines just before the dao insert call?
```
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
        dar.getData().setCreateDate(now.getTime());
        dar.getData().setPartialDarCode("temp_DAR_" + sdf.format(now));
```

Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java]: While we're at it, we could abstract `"temp_DAR_"` into a public static string so both classes can refer to it.

**Review by rushtong — COMMENTED** [review]
Some minor changes requested, but looks good, nice catch on this bug!

File: src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java (hunk 1 above)
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java]: Still need this one change:
```suggestion
        document.put(DarConstants.SORT_DATE, d.getSortDate());
```

**Review by rushtong — CHANGES_REQUESTED** [review]
Thank you for the changes! Still one change to make, see inline.

**Review by rushtong — APPROVED** [review]
Thank you!

**Comment by rushtong** [comment]
@JVThomas Have run `mvn clean package` locally on this branch and it runs fine. Please ignore the current maven action failure, it is a false negative. Merge when you are ready.

**Comment by JVThomas** [comment]
Note: At time of merge there were issues obtaining maven packages (most likely external issue), resulting in the failure of the Coverage and Maven jobs. Code was able to perform locally, indicating that these failures are likely to be false positives.

**Pull request merged**